ussologin/ussologin.go

```diff
@@ -157,5 +162,60 @@ func (f *FileTokenStore) Get() (*usso.SSOData, error) {
 	if err := json.Unmarshal(data, &tok); err != nil {
 		return nil, errgo.Notef(err, "cannot unmarshal token")
 	}
+	if tok.TokenName != name {
+		return nil, nil
+	}
+	return &tok, nil
+}
+
+// DirTokenStore implements the TokenStore interface by storing the
+// JSON-encoded oauth tokens in a directory.
+type DirTokenStore struct {
+	path   string
+	prefix string
+}
+
+// NewDirTokenStore returns a new DirTokenStore that uses the given path
+// for storage. All token files will be have the name prefix + tokenName.
+func NewDirTokenStore(path, prefix string) *DirTokenStore {
+	return &DirTokenStore{
+		path:   path,
+		prefix: prefix,
+	}
+}
+
+// Put implements TokenStore.Put by writing the token to the
+// DirTokenStore's directory.
+func (s *DirTokenStore) Put(tok *usso.SSOData) error {
+	if err := os.MkdirAll(s.path, 0700); err != nil {
+		return errgo.Notef(err, "cannot create directory")
+	}
+	data, err := json.Marshal(tok)
+	if err != nil {
+		return errgo.Notef(err, "cannot marshal token")
+	}
+	if err := ioutil.WriteFile(filepath.Join(s.path, s.prefix+tok.TokenName), data, 0600); err != nil {
+		return errgo.Notef(err, "cannot write file")
+	}
+	return nil
+}
+
+// Get implements TokenStore.Get by
+// reading the token from the DirTokenStore's directory.
+func (s *DirTokenStore) Get(name string) (*usso.SSOData, error) {
+	data, err := ioutil.ReadFile(filepath.Join(s.path, s.prefix+name))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
```

Comment:
I'm not sure it's right that this returns nil if the token's not found.
I think it should return some kind of not-found error
(and that could be documented in the tokenstore interface).